modules/dbparser/patterndb.c

```diff
@@ -313,7 +311,7 @@ _execute_action_create_context(PatternDB *db, PDBProcessParams *process_params)
   PDBRule *rule = process_params->rule;
   PDBContext *triggering_context = process_params->context;
   LogMessage *triggering_msg = process_params->msg;
-  GString *buffer = process_params->buffer;
+  GString *buffer = g_string_sized_new(256);
```

Comment:
Similarly here: stolen string leaks. (It is not replaced with scratchbuffers in the end.)

Reply:
fixed.